Sources/Shared/Models/VIMVideo.m

```diff
@@ -541,8 +541,24 @@ - (NSInteger)likesCount
 - (NSInteger)commentsCount
 {
     VIMConnection *commentsConnection = [self connectionWithName:VIMConnectionNameComments];
-    
-    return (self.canViewComments ? commentsConnection.total.intValue : 0);
+
+    if (self.canViewComments)
+    {
+        if ([commentsConnection.total respondsToSelector:@selector(intValue)])
+        {
+            return commentsConnection.total.intValue;
+        }
+        else
+        {
+            NSAssert(false, @"Error: Detected instance where `NSNumber` is not `NSNumber`.");
```

Comment:
nit pick - "...where `commentsConnection.total` is unexpectedly not an `NSNumber`"